pkg/cloudprovider/types/types.go

```diff
@@ -76,6 +76,11 @@ type Provider interface {
 	SetMetricsForMachines(machines clusterv1alpha1.MachineList) error
 }
 
+// NetworkConfig holds information about cluster networking.
+type NetworkConfig struct {
```

Comment:
I am not sure why do we need this change here? I mean why do we need to specify this via command line? eventually what needs to be done is, deriving this from the machine deployment from here: 
https://github.com/kubermatic/machine-controller/blob/master/pkg/providerconfig/types/types.go#L108-L112

Reply:
[This]( https://github.com/kubermatic/machine-controller/blob/master/pkg/providerconfig/types/types.go#L108-L112) holds machine specific static network configuration. 

Machine doesn't need to know about pod cidrs. It only needs to have IPv6 address so that it can host pods with IPv6 addresses. So we decided not to put it in there.